eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1259,39 +1266,77 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
+            // they accepted the last fee we sent them, so we close without sending a closing_signed
             handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx))))
-          } else if (nextClosingFee == remoteClosingFee) {
-            // we have converged!
-            val closingTxProposed1 = (d.closingTxProposed: @unchecked) match {
-              case previousNegotiations :+ currentNegotiation => previousNegotiations :+ (currentNegotiation :+ ClosingTxProposed(closingTx, closingSigned))
-            }
-            handleMutualClose(signedClosingTx, Left(d.copy(closingTxProposed = closingTxProposed1, bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSigned
+          } else if (d.closingTxProposed.flatten.size >= MAX_NEGOTIATION_ITERATIONS) {
+            // there were too many iterations, we stop negotiating and accept their fee
+            log.warning("could not agree on closing fees after {} iterations, accepting their closing fees ({})", MAX_NEGOTIATION_ITERATIONS, remoteClosingFee)
+            handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSignedRemoteFees
+          } else if (lastLocalClosingSigned_opt.flatMap(_.localClosingSigned.feeRange_opt).exists(r => r.min <= remoteClosingFee && remoteClosingFee <= r.max)) {
+            // they chose a fee inside our proposed fee range, so we close and send a closing_signed for that fee
+            handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSignedRemoteFees
+          } else if (d.commitments.localCommit.spec.toLocal == 0.msat) {
+            // we have nothing at stake so there is no need to negotiate, we accept their fee right away
+            handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSignedRemoteFees
           } else {
-            log.info("proposing closingFeeSatoshis={}", closingSigned.feeSatoshis)
-            val closingTxProposed1 = (d.closingTxProposed: @unchecked) match {
-              case previousNegotiations :+ currentNegotiation => previousNegotiations :+ (currentNegotiation :+ ClosingTxProposed(closingTx, closingSigned))
+            c.feeRange_opt match {
+              case Some(ClosingSignedTlv.FeeRange(minFee, maxFee)) if !d.commitments.localParams.isFunder =>
+                // if we are fundee and they proposed a fee range, we pick a value in that range and they should accept it without further negotiation
+                // we don't care much about the closing fee since they're paying it (not us) and we can use CPFP if we want to speed up confirmation
+                // but we need to ensure it's high enough to at least propagate across the network
+                val txPropagationFeerate = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(1008) * 2
```

Comment:
We do, but we probably want something a bit higher than that though, don't we? I don't want to use directly that value because it could be very very low while the actual feerate is high, that's why I'd rather use an explicit block target.

I chose a far-away block target, and doubled that, but that was completely arbitrary, we can use something else (for example a user-configured block target?).

Reply:
We should let our (funder) peer be quite free to choose a low fee here, so I updated this to the 1008 block target.
I decided not to make it user-configurable, because it would actually be a dick move if they set it to a high value, the spec says you should be tolerant with what fee the funder wants to use.